builder.md

```diff
@@ -67,6 +67,85 @@ These files have the following structure:
       base0D: "dddddd"
       base0E: "eeeeee"
       base0F: "ffffff"
+    overrides:
```

Comment:
Out of curiosity, how did we arrive at the current names? This is a bit of a larger question, but what would we think of organizing this by category?

Stuff like `code.comments`, `code.boolean`, etc.

I realize there's the potential for a lot of bikeshedding here, so as long as we understand where the current names come from, and have good ideas on how to name new names I don't think we need to worry about it too much.

Reply:
I went through the styling spec and used the names used in the "Text Editor" column.
![image](https://github.com/user-attachments/assets/dc2f28ad-8c66-422f-a1c2-4af9e04dcf82)

If we decide on creating things like `code.comments`, etc then I'll probably use TmTheme files as heavy inspiration.

Reply:
How well does yaml cope with code.comments as a key? Or are we suggesting something like 

```yml
...

code:
  comments: "ffffff"
```

If the latter then my preference would be to keep a flatter yml structure to simplify builders

Reply:
Yeah suggesting the latter, mustache doesn't allow property names with dots in them though, do this would be a problem. The flat structure makes sense when we don't have many variables, but when we have many we start prepending the variable names with what could be used as structure instead. As long as we use something like `-` dashes to signify a property of some other object, then to me it's the same. If we signify properties with dashes and then also allow dashes in the property name then it becomes confusing. For example `scheme-supported-systems` isn't clear which are properties. Could be `scheme-supported` with property `system` at a glance for example. So I'd opt for `scheme-supportedSystem` which maintains the child/parent relationship.

Reply:
Yeah I think that's reasonable tbh, I'm quite a fan of camel case. There is absolutely a case for more deeply nested structures but I think it would enable a lot more scope creep in terms of the number of variables used. Themes like textmates and vscodes appear to just grow exponentially in terms of complexity. I think a lot of the appeal of tinted theming is the simplicity in that regard

Reply:
I typed the response without seeing yours first @FredHappyface. 

At base16/base24's core they are simple, most people won't use these override options, but I think it's good to provided the options to allow template maintainers the option to fix bugs with a scheme [or all schemes](https://github.com/tinted-theming/home/issues/10). So while a typical scheme wouldn't use the overrides, I can see specific themes doing something like this: 

```yaml
system: "base16"
name: "Nord"
author: "arcticicestudio"
variant: "dark"
palette:
  base00: "2E3440"
  base01: "3B4252"
  base02: "434C5E"
  base03: "4C566A"
  base04: "D8DEE9"
  base05: "E5E9F0"
  base06: "ECEFF4"
  base07: "8FBCBB"
  base08: "BF616A"
  base09: "D08770"
  base0A: "EBCB8B"
  base0B: "A3BE8C"
  base0C: "88C0D0"
  base0D: "81A1C1"
  base0E: "B48EAD"
  base0F: "5E81AC"
  overrides:
    constant:
      language:
        boolean: "base0C"
  entity: 
    name:
      variable: "base0D"
```

I feel like providing additional options doesn't make it complex. The difference between this and sublime/vscode/tmthemes is that you don't **need** to style any of those variables where in the mentioned themes it's a **requirement** so base16/24 simplicity is default with the override options.

Reply:
Yeah I agree with that. Thanks :)

Reply:
Based on the conversations in Matrix, I've simplified the above list, but only kept items specified in the styling.md document for now:

```yaml
comment: base0x
string:
  quoted: base0x
  regexp: base0x
constant:
  language:
    boolean: base0x
  numeric: base0x
  character:
    entity: base0x
storage:
  type:
    class: base0x
entity:
  name:
    class: base0x
    function: base0x
    tag: base0x
    variable: base0x
  other:
    attributeName: base0x
keyword:
  control: base0x
  declaration: base0x
variable:
  function: base0x
  language: base0x
  other:
    constant: base0x
    property: base0x
markup: base0x
  bold: base0x
  code: base0x
  italic: base0x
  quote: base0x
diff:
  added: base0x
  changed: base0x
  deleted: base0x
  header: base0x
ui:
  background: base0x
  backgroundDark: base0x
  backgroundLight: base0x
  deprecated: base0x
  foreground: base0x
  foregroundDark: base0x
  foregroundLight: base0x
  lineBackground: base0x
  searchText: base0x
  selectionBackground: base0x
```